core/extension/extension_api_dump.cpp

```diff
@@ -938,9 +939,9 @@ void NativeExtensionAPIDump::generate_extension_json_file(const String &p_path)
 	Ref<JSON> json;
 	json.instantiate();
 
-	String text = json->stringify(api, "\t", false);
+	String text = json->stringify(api, "\t", false) + "\n";
 	Ref<FileAccess> fa = FileAccess::open(p_path, FileAccess::WRITE);
-	CharString cs = text.ascii();
-	fa->store_buffer((const uint8_t *)cs.ptr(), cs.length());
+	fa->store_string(text);
```

Comment:
Forcing ascii might have been intentional, if so I can revert that change. I first thought storing the buffer like this was the cause for the missing newline but `store_string` does it the same way, just with `utf8()` instead of `ascii()`.

Reply:
`store_line(...)` add a new line on the end, `store_string(...)` do not (no other differences).

I guess UTF-8 is fine, currently all identifiers should be ASCII, but forcing ASCII should not be necessary.